### ISSUE TITLE
Documentation/Method unclear for Pretrained File in Embedding

### ISSUE THREAD

**Issue by mayhewsw** [allenai/allennlp#2898]
TLDR; The code for programmatically using pretrained embedding file in the Embedding object is hard to understand and use. 

The [official tutorial](https://allennlp.org/tutorials) uses AllenNLP programmatically, and includes embeddings, but not from pretrained. I tried implementing my own model programmatically, and used Embedding(pretrained_file="/path/to/glove", ...) but found that this does nothing. 

AFAIK, the way to do it is to use a private method ("_read_pretrained_embeddings_file") and pass the resultant weights to the Embedding object (argument "weight"). 

This is counterintuitive. I expected that the "pretrained_file" argument would work, because that's how the config files work.

**Comment by kernelmachine** [comment]
The way to use the `Embedding` module independently is through `from_params`, which is also how the config files access it. 

More info here: 
https://github.com/allenai/allennlp/blob/master/allennlp/modules/token_embedders/embedding.py#L75

**Comment by mayhewsw** [comment]
Thanks. The comment is explicit. If only I'd seen that at the beginning...  

But my point is that this is a strange workflow: if you want to do this programmatically (which is not unreasonable), then you have to create params, and load them. It seems like more sensible arguments to Embedding would be helpful. This could be a taste issue though.

**Comment by matt-gardner** [comment]
I definitely agree that there is something suboptimal about all of this.  I'm not sure what the right way is to fix it in an backwards-compatible way.  Suggestions welcome.

**Comment by bratao** [comment]
@matt-gardner this bite me just now. With the 1.0 refactoring would not be good to use the pretrained_file argument to effectively load the pretrained file?

**Comment by DeNeutoy** [comment]
@brato, thanks for re-raising this issue, it is a terrible API - we will fix it in 1.0 👍 

I think we can just move this block inside the constructor:

https://github.com/allenai/allennlp/blob/master/allennlp/modules/token_embedders/embedding.py#L328-L336


Such that the only thing the embedding `from_vocab_or_files` classmethod does is check the size of the embedding from the vocab. Hopefully should be an easy change, I don't see too much wrong with this approach.

**Comment by bratao** [comment]
This makes perfect sense @DeNeutoy . Thank you for planning to fix in 1.0
